pkg/pipeline/encode/encode_prom.go

```diff
@@ -176,53 +249,67 @@ func NewEncodeProm() (Encoder, error) {
 
 	portNum := jsonEncodeProm.Port
 	promPrefix := jsonEncodeProm.Prefix
+	expiryTime := int64(jsonEncodeProm.ExpiryTime)
+	if expiryTime == 0 {
+		expiryTime = defaultExpiryTime
+	}
+	log.Debugf("expiryTime = %d", expiryTime)
 
-	counters := make(map[string]counterInfo)
-	gauges := make(map[string]gaugeInfo)
-	histograms := make(map[string]histInfo)
-	for _, metricInfo := range jsonEncodeProm.Metrics {
-		fullMetricName := promPrefix + metricInfo.Name
-		labels := metricInfo.Labels
+	metrics := make(map[string]metricInfo)
+	for _, mInfo := range jsonEncodeProm.Metrics {
+		var pMetric PromMetric
+		fullMetricName := promPrefix + mInfo.Name
+		labels := mInfo.Labels
 		log.Debugf("fullMetricName = %v", fullMetricName)
-		log.Debugf("labels = %v", labels)
-		switch metricInfo.Type {
+		log.Debugf("Labels = %v", labels)
+		pMetric.metricType = mInfo.Type
+		switch mInfo.Type {
 		case api.PromEncodeOperationName("Counter"):
 			counter := prometheus.NewCounterVec(prometheus.CounterOpts{Name: fullMetricName, Help: ""}, labels)
 			err := prometheus.Register(counter)
```

Comment:
I am looking on https://github.com/netobserv/flowlogs2metrics/pull/33/files#diff-30d11303a4ad1ec82e4f39e3ae6dc7d044c8dd370c536a98ba90febbfe2d425aR277 and on https://github.com/netobserv/flowlogs2metrics/pull/33/files#diff-30d11303a4ad1ec82e4f39e3ae6dc7d044c8dd370c536a98ba90febbfe2d425aR269 and on https://github.com/netobserv/flowlogs2metrics/pull/33/files#diff-30d11303a4ad1ec82e4f39e3ae6dc7d044c8dd370c536a98ba90febbfe2d425aR286 can you collapse this code so it will not be repeating in each of the cases ???